cobc/tree.c

```diff
@@ -6048,12 +6048,12 @@ cb_build_binary_op (cb_tree x, const enum cb_binary_op_op op, cb_tree y)
 		rel_bin_op = 1;
 		if ((CB_REF_OR_FIELD_P (x))
 		 && CB_FIELD_PTR (x)->level == 88) {
-			cb_error_x (e, _("invalid expression"));
+			cb_error_x (e, _("invalid expression: conditional on the left of numeric operator"));
```

Comment:
So, I tried, but it does not look as easy as expected, as it triggers another error before this one (`condition-name not allowed here`). After a few tries, I found another error that segfaulted the compiler in `decimal_expand()`, hence another fix in this PR.

Reply:
resolved by disabling this check and drop the gettextize from the message that would otherwise result in a the new msgid; it does not look like we have a testcase for condition-name in expressions, so please add that to your test, then we can mark this as resolved